lib/facter/util/aix/info_extractor.rb

```diff
@@ -6,17 +6,71 @@ module Aix
       module InfoExtractor
         MEGABYTES_EXPONENT = 1024**2
         GIGABYTES_EXPONENT = 1024**3
+        PROPERTIES_LSPV = [
+          'PHYSICAL VOLUME:',
+          'VOLUME GROUP:',
+          'PV IDENTIFIER:',
+          'VG IDENTIFIER',
+          'PV STATE:',
+          'STALE PARTITIONS:',
+          'ALLOCATABLE:',
+          'PP SIZE:',
+          'LOGICAL VOLUMES:',
+          'TOTAL PPs:',
+          'VG DESCRIPTORS:',
+          'FREE PPs:',
+          'HOT SPARE:',
+          'USED PPs:',
+          'MAX REQUEST:',
+          'FREE DISTRIBUTION:',
+          'USED DISTRIBUTION:',
+          'MIRROR POOL:'
+        ].freeze
+        PROPERTIES_LSLV = [
+          'LOGICAL VOLUME:',
+          'VOLUME GROUP:',
+          'LV IDENTIFIER:',
+          'PERMISSION:',
+          'VG STATE:',
+          'LV STATE:',
+          'TYPE:',
+          'WRITE VERIFY:',
+          'MAX LPs:',
+          'PP SIZE:',
+          'COPIES:',
+          'SCHED POLICY:',
+          'LPs:',
+          'PPs:',
+          'STALE PPs:',
+          'BB POLICY:',
+          'INTER-POLICY:',
+          'RELOCATABLE:',
+          'INTRA-POLICY:',
+          'UPPER BOUND:',
+          'MOUNT POINT:',
+          'LABEL:',
+          'MIRROR WRITE CONSISTENCY:',
+          'EACH LP COPY ON A SEPARATE PV ?:',
+          'Serialize IO ?:'
+        ].freeze
 
-        def self.extract(content, regex)
-          content = content.each_line.map do |line|
-            next unless regex =~ line
-
-            line.split(/:\s*|\s{2,}/)
+        def self.extract(content, cmd)
+          property_hash = {}
+          properties = case cmd
+                       when :lspv
+                         PROPERTIES_LSPV
+                       when :lslv
+                         PROPERTIES_LSLV
+                       end
```

Comment:
can also have a single `PROPERTIES` constant that is a hash of type `PROPERTIES = { :lspv => <lspv_properties>, :lslv => <lslv_properties> }` then you can rewrite this line as `properties = PROPERTIES[cmd]`